server/src/main/java/com/thoughtworks/go/server/controller/ArtifactsController.java

```diff
@@ -91,73 +97,63 @@
-                                          @RequestParam("buildName") String buildName,
-                                          @RequestParam("filePath") String filePath,
-                                          @RequestParam(value = "sha1", required = false) String sha
-    ) throws Exception {
-        return getArtifact(filePath, jsonViewFactory, pipelineName, counterOrLabel, stageName, stageCounter, buildName, sha, null);
-    }
-
-    @RequestMapping(value = "/repository/restful/artifact/GET/zip", method = RequestMethod.GET)
-    public ModelAndView getArtifactAsZip(@RequestParam("pipelineName") String pipelineName,
-                                         @RequestParam("pipelineLabel") String counterOrLabel,
-                                         @RequestParam("stageName") String stageName,
-                                         @RequestParam(value = "stageCounter", required = false) String stageCounter,
-                                         @RequestParam("buildName") String buildName,
-                                         @RequestParam("filePath") String filePath,
-                                         @RequestParam(value = "sha1", required = false) String sha
-    ) throws Exception {
-        return getArtifact(filePath, zipViewFactory, pipelineName, counterOrLabel, stageName, stageCounter, buildName, sha, null);
-    }
-
-    @RequestMapping(value = "/repository/restful/artifact/GET/*", method = RequestMethod.GET)
-    public void fetch(HttpServletRequest request, HttpServletResponse response) throws Exception {
-        request.getRequestDispatcher("/repository/restful/artifact/GET/html").forward(request, response);
+    @RequestMapping(value = {
+            "/files/{pipelineName}/{pipelineLabel}/{stageName}/{stageCounter}/{buildName:.+}",
+            "/files/{pipelineName}/{pipelineLabel}/{stageName}/{stageCounter}/{buildName}/*",
+            "/files/{pipelineName}/{pipelineLabel}/{stageName}/{stageCounter}/{buildName}/**"
+    }, method = RequestMethod.GET)
+    public ModelAndView getArtifacts(@PathVariable("pipelineName") String pipelineName,
+                                     @PathVariable("pipelineLabel") String counterOrLabel,
+                                     @PathVariable("stageName") String stageName,
+                                     @PathVariable(value = "stageCounter", required = false) String stageCounter,
+                                     @PathVariable("buildName") String buildName,
+                                     @RequestParam(value = "sha1", required = false) String sha,
+                                     @RequestParam(value = "serverAlias", required = false) String serverAlias,
+                                     @RequestParam(value = "startLineNumber", required = false) Long start,
+                                     HttpServletRequest request) throws Exception {
+        if (start == null) {
```

Comment:
in earlier implementation, it was done using filename `consoleout.json`. I think is better to keep it that way only.